dune/http.go

```diff
@@ -13,6 +16,88 @@ type ErrorResponse struct {
 	Error string `json:"error"`
 }
 
+type RateLimit struct {
+	Limit     int
+	Remaining int
+	Reset     int64
+}
+
+type APIError struct {
+	StatusCode  int
+	StatusText  string
+	BodySnippet string
+	RateLimit   *RateLimit
+	RetryAfter  time.Duration
+}
+
+func (e *APIError) Error() string {
+	if e.BodySnippet != "" {
+		return fmt.Sprintf("http %d %s: %s", e.StatusCode, e.StatusText, e.BodySnippet)
+	}
+	return fmt.Sprintf("http %d %s", e.StatusCode, e.StatusText)
+}
+
+type RetryPolicy struct {
+	MaxAttempts           int
+	InitialBackoff        time.Duration
+	MaxBackoff            time.Duration
+	Jitter                time.Duration
+	RetryableStatusCodes  []int
+}
+
+var defaultRetryPolicy = RetryPolicy{
+	MaxAttempts:          3,
+	InitialBackoff:       500 * time.Millisecond,
+	MaxBackoff:           5 * time.Second,
+	Jitter:               100 * time.Millisecond,
+	RetryableStatusCodes: []int{429, 500, 502, 503, 504},
+}
+
+func parseRateLimitHeaders(h http.Header) *RateLimit {
+	limStr := h.Get("X-RateLimit-Limit")
+	remStr := h.Get("X-RateLimit-Remaining")
+	resetStr := h.Get("X-RateLimit-Reset")
+
+	var lim, rem int
```

Comment:
No need to shorten variable names :) 

```suggestion
	var limit, remaining int
```